packages/libs/error-stack/src/fmt/mod.rs

```diff
@@ -3,20 +3,19 @@
 //!
 //! # Hooks
 //!
-//! The format implementation (especially the [`Debug`] implementation),
-//! can be easily extended using hooks.
-//! Hooks are functions of the signature `Fn(&T, &mut HookContext<T>)`, they provide an easy and
-//! ergonomic way to partially modify the format and enable the output of types that are
-//! not necessarily added via [`Report::attach_printable`] or are unable to implement [`Display`].
+//! The [`Debug`] implementation can be easily extended using hooks. Hooks are functions of the
+//! signature `Fn(&T, &mut HookContext<T>)`, they provide an easy and ergonomic way to partially
+//! modify the format and enable the output of types that are not necessarily added via
+//! [`Report::attach_printable`] or are unable to implement [`Display`].
 //!
 //! Hooks can be attached through the central hooking mechanism which `error-stack`
 //! provides via [`Report::install_debug_hook`].
 //!
 //! Hooks are called for contexts which provide additional values through [`Error::provide`] or
-//! [`Context::provide`] and attachments which are added via [`Report::attach`].
-//! For contexts and values, that can be requested using [`Frame::request_ref`] and
-//! [`Frame::request_value`], the rendering order is determined by the order of
-//! [`Report::install_debug_hook`] calls.
+//! [`Context::provide`] and attachments which are added via [`Report::attach`] or
+//! [`Report::attach_printable`]. For contexts and values, that can be requested using
+//! [`Frame::request_ref`] and [`Frame::request_value`], the rendering order is determined by the
+//! order of [`Report::install_debug_hook`] calls.
```

Comment:
```suggestion
//! [`Report::attach_printable`]. The order of [`Report::install_debug_hook`] calls determines the order of the rendered output.
```

The sentence was pretty unwieldy. I think we should try and make it clearer somewhere else that if a `Context` can only be gotten by downcasting, then the hook isn't called for it. Rather than putting it in this confusing sentence 😁

Reply:
Addressed in 7707b0dbf8fd8cda72e21bf76a3978bb2568afe2